bestbook/views/__init__.py

```diff
@@ -125,11 +125,24 @@ class (e.g. Ask, Login, Observe, Submit)
             "submit": Submit,
         }
         form = request.form
+
         return jsonify(forms[resource]().post())
 
 # API POST Handlers
 
+def require_auth(func):
+    def authed_func(*args, **kwargs):
+        s3_keys = {
+            'access': request.data.get('s3_access'),
+            'secret': request.data.get('s3_secret'),
```

Comment:
This is failing when I try to submit a recommendation.  This is a flask.Request, and its data is a bytestring instead of a dictionary.  Maybe it would be better the keys were in the headers?